### PR TITLE
detect-id: convert unittests to FAIL/PASS APIs

### PR DESCRIPTION
**Pull request by TheKharleeci**
Make sure these boxes are signed before submitting your Pull Request -- thank you.

- [x] I have read the contributing guide lines at https://redmine.openinfosecfoundation.org/projects/suricata/wiki/Contributing
- [x] I have signed the Open Information Security Foundation contribution agreement at https://suricata-ids.org/about/contribution-agreement/
- [ ] I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)

Link to redmine ticket:
https://redmine.openinfosecfoundation.org/issues/4046

Ticket: #4046

Previous PR: #6619 

Describe changes:

- Update detect-id unit tests to use FAIL/PASS APIs
- Update Copyright Year

### CODE REVIEW

File: src/detect-id.c [hunk 1]
```diff
@@ -370,26 +364,9 @@ static int DetectIdTestMatch01(void)
...
-    uint32_t results[3][3] = {
-                              /* packet 0 match sid 1 but should not match sid 2 */
-                              {1, 0, 0},
-                              /* packet 1 should not match */
-                              {0, 1, 0},
-                              /* packet 2 should not match */
-                              {0, 0, 1} };
-
-    result = UTHGenericTest(p, 3, sigs, sid, (uint32_t *) results, 3);
-
```
Comment (jufajardini) [inline src/detect-id.c:389]: If we deleted these, we would be deleting the core of this test ;)

File: src/detect-id.c [hunk 2]
```diff
@@ -370,26 +364,9 @@ static int DetectIdTestMatch01(void)
...
-    uint32_t results[3][3] = {
-                              /* packet 0 match sid 1 but should not match sid 2 */
-                              {1, 0, 0},
-                              /* packet 1 should not match */
-                              {0, 1, 0},
-                              /* packet 2 should not match */
-                              {0, 0, 1} };
-
-    result = UTHGenericTest(p, 3, sigs, sid, (uint32_t *) results, 3);
```
Comment (jufajardini) [inline src/detect-id.c:388]: I think a check like 
```c
FAIL_IF_NOT(UTHGenericTest(p, 3, sigs, sid, (uint32_t *) results, 3));
```
here would work :)

**Review by jufajardini — CHANGES_REQUESTED** [review]
Looking good, except for `DetectIdTestMatch01` which needs some adjustment :)

**Comment by TheKharleeci** [comment]
Followed by: #6626

**Pull request closed without merging**